### PR TITLE
aarch64 port changes

### PR DESCRIPTION
**Pull request by RamaMalladiAWS**
This PR is for porting Velox to run an aarch64 based machine.

### CODE REVIEW

File: velox/common/base/SimdUtil.h [hunk 1]
```diff
@@ -354,7 +354,12 @@ xsimd::batch<T, A> iota(const A& = {});
 template <typename T, typename A = xsimd::default_arch>
 xsimd::batch<T, A> setAll(T value, const A& = {}) {
   if constexpr (std::is_same_v<T, bool>) {
+#if defined(__aarch64__)
+    return xsimd::batch<T, A>(
+        xsimd::broadcast<bool, A>((bool)(value ? -1 : 0)));
```
Comment (Yuhta) [inline velox/common/base/SimdUtil.h:359]: As we discussed in #3487 let us put a test to make sure the bits are all set:

```c++
auto bits = setAll(true);
auto words = reinterpret_cast<int64_t*>(&bits);
for (int i = 0; i < xsimd_batch<int64_t>::size; ++i) {
  EXPECT_EQ(words[i], -1ll);
}
```

Comment (RamaMalladiAWS) [reply velox/common/base/SimdUtil.h:359]: Thanks @Yuhta. I will add this unit test and confirm if my implementation is correct or not.

Comment (RamaMalladiAWS) [reply velox/common/base/SimdUtil.h]: Fixed the implementation and uploaded the code. Thank you.

File: velox/exec/tests/XsimdBatchSetAllTest.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+#include "velox/exec/tests/XsimdBatchSetAllTest.h"
+#include "velox/common/base/tests/GTestUtils.h"
+
+#ifndef XSIMD_NO_SUPPORTED_ARCHITECTURE
+using namespace facebook::velox::simd;
+class XsimdBatchTest : public testing::Test {
+ protected:
+  void TestSetAll() const {
```
[the remark this answers is not in the document]
Comment (RamaMalladiAWS) [reply velox/exec/tests/XsimdBatchSetAllTest.cpp:24]: As in rename the file to SimdUtilTest.cpp?

[the remark this answers is not in the document]
Comment (Yuhta) [reply velox/exec/tests/XsimdBatchSetAllTest.cpp:24]: No there is already a file called `SimdUtilTest.cpp` in `common/base/tests`

[the remark this answers is not in the document]
Comment (RamaMalladiAWS) [reply velox/exec/tests/XsimdBatchSetAllTest.cpp]: Yes, got it. Will do the change and push it. Thanks

**Review by RamaMalladiAWS — COMMENTED** [review]
moved unittest for setAll into SimdUtilTest.cpp

File: velox/common/base/tests/SimdUtilTest.cpp [hunk 3]
```diff
@@ -107,9 +108,21 @@ class SimdUtilTest : public testing::Test {
     EXPECT_EQ(reference, target);
   }
 
+  void testSetAll() const {
```
Comment (Yuhta) [inline velox/common/base/tests/SimdUtilTest.cpp:111]: No need for creating a method in fixture.  Just create the test case and put the content there.

File: velox/common/base/tests/SimdUtilTest.cpp [hunk 4]
```diff
@@ -20,6 +20,7 @@
 #include <gtest/gtest.h>
 
 using namespace facebook::velox;
+using namespace facebook::velox::simd;
```
Comment (Yuhta) [inline velox/common/base/tests/SimdUtilTest.cpp:23]: Just call `simd::setAll` in the test case.

Comment (RamaMalladiAWS) [reply velox/common/base/tests/SimdUtilTest.cpp]: Done

**Review by RamaMalladiAWS — COMMENTED** [review]
more changes to SimdUtilTest.cpp

**Review by Yuhta — APPROVED** [review]
Looks good to me now.  Let me know when the CLA is ready.

**Comment by RamaMalladiAWS** [comment]
> @Yuhta has imported this pull request. If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42196145).

Thank you @Yuhta for importing the pull request. I see "some checks were not successful". Do I have any action here? Thanks

**Comment by Yuhta** [comment]
@RamaMalladiAWS No worry just unrelated warnings

**Comment by RamaMalladiAWS** [comment]
> @RamaMalladiAWS No worry just unrelated warnings

Thank you.

**Pull request closed without merging**